axiom/nr_log_events.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_axiom.h"
+
+#include <math.h>
+#include <stdlib.h>
+
+#include "nr_log_events.h"
+#include "nr_analytics_events_private.h"
+#include "util_memory.h"
+
+bool nr_log_events_add_event(nr_analytics_events_t* events,
+                             const nr_log_event_t* event,
+                             nr_random_t* rnd) {
+  bool events_sampled = false;
+  char* log_event_json = NULL;
+  nr_analytics_event_t* log_event;
+
+  if (NULL == events) {
+    return false;
```

Comment:
the return value seems a little overloaded.  it both indicates if sampling occurred and if an error adding the event occurred. Should we clarify that here: https://github.com/newrelic/newrelic-php-agent/pull/500/files#diff-e5f4823dae53cfaeb71612eec0542c18feef8bc10115a585516b1fe487bd2a35R18

Reply:
@zsistla These `NULL` checks are more of an asserts (check if the arguments are valid pointers) rather than business logic checks (i.e. has sampling occurred?). If an event was not added because invalid arguments were passed to this function (event != NULL is ensured by the caller [here](https://github.com/misiekn/newrelic-php-agent/blob/53d5617ed81910c5e007485fd821b60cc0356f04/axiom/nr_txn.c#L3369)), sampling could not have occurred and therefore `false` is returned. `true` is returned if and only if sampling occurs. It's impossible not to overload `false` return value with this design of the function (failing check for `NULL` arguments needs to return something) however the interesting value (`true`) is returned only if sampling occurs and this is the intention of this function: return `true` if sampling occurred, `false` otherwise (sampling did not occur or event was not added, which implies sampling must not have occurred).

Reply:
Function's doc string updated in f426ac1f41bf07ec0bd24851078ce944edafd680